### PR TITLE
wasm codegen fixes 

### PR DESCRIPTION
**Pull request by xxxbxxx**
```zig
test {
    var a: i32 = -12345;
    var b: i64 = -12345;
    const c: i21 = @intCast(b);
    const d: i60 = c;
    std.debug.print("{} {} {} {}\n", .{c, b, a, d});
}
``` 

previously: 
> `12345 -53017076301825 -12345 4539230677105560`

now:
> `-12345 -12345 -12345 -12345` 

in addition, enable more tests: 1543 -> 1611 passed

### CODE REVIEW

File: test/behavior/stage2_wasm_div.zig [hunk 1]
```diff
@@ -6,7 +6,6 @@ test "wasm integer division" {
     // This test is copied from int_div.zig, with additional test cases for @divFloor on floats.
     // TODO: Remove this test once the division tests in math.zig and int_div.zig pass with the
     // stage2 wasm backend.
-    if (builtin.zig_backend != .stage2_wasm) return error.SkipZigTest;
```
Comment (Luukdegram) [inline test/behavior/stage2_wasm_div.zig:9]: This file can be removed as mentioned by the TODO. Right now this change enables the test for all backends instead.

**Review by Luukdegram — CHANGES_REQUESTED** [review]
Great work on those fixes! Apart from the one comment, it should be good to go.

I do have a request for future contributions: Please refrain from enabling behavior tests that are not affected by the fixes/improvements. Considering I have to verify all the tests myself to ensure there are no false positives it would be a lot easier to review a PR without this as I could do it myself anyway. I do understand that it helps to find out what you could work on, so once I'm healthy again I'll try to keep them up-to-date. I appreciate the time and effort you put into these, so let's enable those tests now.

**Comment by xxxbxxx** [comment]
> Great work on those fixes! Apart from the one comment, it should be good to go.
> 
> I do have a request for future contributions: Please refrain from enabling behavior tests that are not affected by the fixes/improvements. Considering I have to verify all the tests myself to ensure there are no false positives it would be a lot easier to review a PR without this as I could do it myself anyway. I do understand that it helps to find out what you could work on, so once I'm healthy again I'll try to keep them up-to-date. I appreciate the time and effort you put into these, so let's enable those tests now.

sure! I'll drop the last commit.

File: test/behavior/stage2_wasm_div.zig (hunk 1 above)
Comment (xxxbxxx) [reply test/behavior/stage2_wasm_div.zig:9]: I think it still doesn't pass. i'll double check

**Comment by Luukdegram** [comment]
> sure! I'll drop the last commit.

No need to drop it this time. You've already put the effort into it and I've verified them locally.

File: test/behavior/stage2_wasm_div.zig (hunk 1 above)
Comment (Luukdegram) [reply test/behavior/stage2_wasm_div.zig:9]: If it doesn't pass yet, we can keep this test case. But I'd suggest not removing the condition as this file was purposely for Wasm only until it passes the original tests.

**Pull request merged**